Name: Bou Haidara
Email: Bouhaidara93@gmail.com

Part 1: College Rankings 

 

Examine the following pages and rankings:

https://www.usnews.com/best-collegesLinks to an external site.

https://www.princetonreview.com/college-rankings/best-collegesLinks to an external site.

https://www.forbes.com/top-colleges/Links to an external site.

Your goal in this part of the assignment is to examine the methodology of these rankings and attempt to recreate it with the scorecard data. Your process should be well justified and presented through the use of markdowns. You don't need to recreate the methodology exactly, especially considering there may be criteria being used that you don't have access to. I would choose, at minimum, the top 5 highly-weighted criteria in the methodology and build your recreation from those.

In [3]:
#i used https://youtu.be/5rNu16O3YNE
#i used https://github.com/vtyeh/pandas-challenge to help me understand how to structure it and ranking
#i followed the example that were in class with my peers using functions to create the listing

import pandas as pd
import numpy as np

#ranking colleges based off of these websites below

#https://www.usnews.com/education/best-colleges/articles/how-us-news-calculated-the-rankings
#https://www.forbes.com/sites/michaeltnietzel/2022/08/30/americas-military-service-academies-an-outstanding-option-for-undergraduate-education/?sh=53e88c196b85
#https://www.forbes.com/sites/emmawhitford/2022/06/09/the-strongest-and-weakest-colleges-in-america---behind-forbes-2022-financial-grades/?sh=3ab89b47ffad

#College Score Card possible factors
#acceptance rate: ADM_RATE, median earing: MD_EARN_WNE_P10, retention rate: RET_FT4, grad rates: C150_4, loans: GRAD_DEBT_MDN_SUPP, fullpaid loans: BBRR2_FED_UG_PAIDINFULL_SUPP
#graduation: C150_4_POOLED_SUPP, student return: RET_FT4_POOLED_SUP, AVGFACSAL, TUITFTE, ADM_RATE_ALL
#grad rates:'C150_4', retention rate:'RET_FT4', median earing:'MD_EARN_WNE_P10', student_fac: 'STUFACR', Admin rate: 'ADM_RATE'

#Readinf the data in and converting all PrivacySuppressed into Nan, I am using .dropna() for the after my function
colleges = pd.read_csv("Most-Recent-Cohorts-Institution.csv",index_col="INSTNM", low_memory=False)
#replacing the PS to NaN
colleges2 = colleges.replace('PrivacySuppressed', np.nan)
#dropping the NaN values
colleges2 = colleges2.dropna(subset=['ADM_RATE','RET_FT4','C150_4','MD_EARN_WNE_P10','GRAD_DEBT_MDN_SUPP','C150_4_POOLED_SUPP', 'TUITFTE', 'AVGFACSAL'])
#colleges2 = colleges2.dropna(subset=['C150_4_POOLED_SUPP', 'RET_FT4_POOLED_SUP', 'AVGFACSAL', 'TUITFTE', 'ADM_RATE_ALL'])



#function for the factors that are choosen
def scoreCard(row):
    #conveting admin rates into a floating (decimal accepting numbers) for var admin
    admin = float(row['ADM_RATE_ALL'])
    #converting retention into a float for var rten
    rten = float(row['RET_FT4']) 
    #converting total graduations into a float for var gard
    grad = float(row['C150_4_POOLED_SUPP'])
    #converting earnigns into a float var earnings
    earnings = float(row['MD_EARN_WNE_P10'])
    #converting debt into a float var debt
    debt = float(row['GRAD_DEBT_MDN_SUPP'])
    #converting current graduations to a float var grad2
    grad2 = float(row['C150_4'])
    #converting student tution help into a float var stu
    stu = float(row['TUITFTE'])
    #converting student fincancal help into a float var stu2
    stu2 = float(row['AVGFACSAL'])
    
    
    #creating the percentages for the values out of .1 (100%)
    uniPercent = {
        'admin': .2,
        'rten' : .2,
        'earnings' : .4,
        'debt' : .05,
        'stu2' : .05,
        'grad' : .1
    }
    
    #scoring system
    uniScore = (
        #using the var uniPercent and multiplying it to itself to make the score a smaller number and adding them up
        uniPercent['admin'] * admin 
        + uniPercent['rten'] * rten 
        + uniPercent['stu2'] * stu2 
        + uniPercent['grad'] * grad
        #below is the going to be the ROI (return of investment) for going to the schools 
        #schools with smaller earnings and higher debt will be way at the bottom 
        / ( uniPercent['earnings'] * earnings 
        - uniPercent['debt'] * debt)
    
    )
   
    #returning the scoring points 
    return uniScore

#making sure the data within the file uses the scoreCard function to tally the points
colleges2['uniScore'] = colleges2.apply(scoreCard, axis = 1)

#creating a var college_score to sore the values from the the file
college_score = colleges2.sort_values(by = 'uniScore', ascending=False)

#var college score are sorted by the scoring system with the top 10 schools
college_score[['uniScore', 'STABBR']].head(10)




uniScore STABBR
INSTNM                                                         
Harvard University                           1057.311363     MA
Stanford University                          1048.731882     CA
Princeton University                         1041.927863     NJ
Columbia University in the City of New York   983.203863     NY
Massachusetts Institute of Technology         982.209802     MA
Yale University                               974.643903     CT
University of California-Los Angeles          964.519923     CA
University of Pennsylvania                    935.307942     PA
University of Chicago                         921.813923     IL
University of California-Berkeley             901.577243     CA

Part 2: New Ranking systems

It should not be a surprise to anyone that there are.. limitations to how these rankings are determined. They have a narrow focus that centers around a predetermined result (the top schools must always remain the top schools). In this section, you will determine your own ranking system using the aspects of colleges that are important to you. In a markdown cell, tell me what factors are important to you and why, then rerun the rankings. Display the new results and contrast them with the results of Part 1.

In [215]:
#https://www.forbes.com/video/6311598410112/the-top-5-colleges-in-the-us/?sh=46787428f2ab
#https://www.forbes.com/pictures/63062f99a431a1d1ebc1d377/top-25-lowest-debt/?sh=4cb895ae261a
#https://www.forbes.com/pictures/63067b2c97683f2a951b5fd4/1-harvard-university/?sh=1e2616cd43f3
#chrome-extension://efaidnbmnnnibpcajpcglclefindmkaj/https://collegescorecard.ed.gov/assets/InstitutionDataDocumentation.pdf

#College Score Card possible factors:
#state: "STABBR","median earing:'MD_EARN_WNE_P10',"GRAD_DEBT_MDN_SUPP",grad rates:'C150_4', retention rate:'RET_FT4'
#acceptance rate: ADM_RATE, median earing: MD_EARN_WNE_P10, retention rate: RET_FT4, grad rates: C150_4, loans: GRAD_DEBT_MDN_SUPP, fullpaid loans: BBRR2_FED_UG_PAIDINFULL_SUPP
#graduation: C150_4_POOLED_SUPP, student return: RET_FT4_POOLED_SUP, AVGFACSAL, TUITFTE, ADM_RATE_ALL
#grad rates:'C150_4', retention rate:'RET_FT4', median earing:'MD_EARN_WNE_P10', student_fac: 'STUFACR', Admin rate: 'ADM_RATE'



#Readind the data in and converting all PrivacySuppressed into Nan, I am using .dropna() for the after my function
mySchool = pd.read_csv("Most-Recent-Cohorts-Institution.csv",index_col="INSTNM", low_memory=False)
#replacing the PS to NaN
mySchool = mySchool.replace('PrivacySuppressed', np.nan)
#Dropping NaN values
mySchool = mySchool.dropna(subset=['ADM_RATE','RET_FT4','BBRR2_FED_UG_PAIDINFULL_SUPP','MD_EARN_WNE_P10'
                                   ,'GRAD_DEBT_MDN_SUPP', 'VETERAN', 'PAR_ED_PCT_1STGEN', 'BBRR2_FED_UG_PAIDINFULL_SUPP',
                                  'TUITFTE'])



#function for the factors that are choosen
def scoreCard2(row):
    #conveting admin rates into a floating (decimal accepting numbers) for var admin
    admin = float(row['ADM_RATE_ALL'])
    #converting retention into a float for var rten
    rten = float(row['RET_FT4']) 
    #converting total graduations into a float for var gard
    grad = float(row['C150_4_POOLED_SUPP'])
    #converting earnigns into a float var earnings
    earnings = float(row['MD_EARN_WNE_P10'])
    #converting debt into a float var debt
    debt = float(row['GRAD_DEBT_MDN_SUPP'])
    #converting current graduations to a float var grad2
    grant = float(row['BBRR2_FED_UG_PAIDINFULL_SUPP'])
    #converting student veteran population help into a float var stu
    stu = float(row['VETERAN'])
    #converting  first-generation on the FAFSA into a float var stu2
    stu2 = float(row['PAR_ED_PCT_1STGEN'])
    #converting student fully repaid debt into a float var stu
    debt2 = float(row['BBRR2_FED_UG_PAIDINFULL_SUPP'])
    #converting  first-generation on the FAFSA into a float var stu2
    stu3 = float(row['TUITFTE'])
    
    
    #creating the percentages for the values out of 1 (100%)
    uniPercent = {
        'admin': .1,
        'rten' : .1,
        'earnings' : .1,
        'debt' : .1,
        'stu2' : .1,
        'grad' : .1,
        'grant' : .1,
        'stu' : .1,
        'debt2' : .1,
        'stu3' : .1
    }
    
    #scoring system
    uniScore2 = (
        uniPercent['admin'] * admin
        + uniPercent['rten'] * rten
        + uniPercent['stu2'] * stu2
        + uniPercent['stu'] * stu
        + uniPercent['stu3'] * stu3
        + uniPercent['grad'] * grad) 
        + 
        #The ROI based on your earning potential and the amount of money/grants you can get
        #with amount of debt you can get subtracted by the ability to pay it off within 8 years
        (uniPercent['earnings'] * earnings + uniPercent['grant'] * grant
        - (uniPercent['debt'] * debt - uniPercent['debt2'] * debt2))

    return uniScore2

#making sure the data within the file uses the scoreCard function to tally the points
mySchool['uniScore2'] = mySchool.apply(scoreCard2, axis = 1)

#creating a var college_score to sore the values from the the file
my_score = mySchool.sort_values(by = 'uniScore2', ascending=False)

#var college score are sorted by the scoring system with the top 10 schools
my_score[['uniScore2', 'STABBR']].head(10)





uniScore2 STABBR
INSTNM                                                                 
University of Pennsylvania                          12984.368596     PA
Columbia University in the City of New York         10886.459674     NY
Northeastern University                              9658.770891     MA
New York University                                  9578.464096     NY
University of California-Berkeley                    8976.197905     CA
University of Michigan-Ann Arbor                     8328.786345     MI
Embry-Riddle Aeronautical University-Daytona Beach   8273.863847     FL
Embry-Riddle Aeronautical University-Prescott        7961.568507     AZ
Drexel University                                    7955.000099     PA
Chamberlain University-Ohio                          7868.234120     OH

The schools located above are the most fincially friendly and veteran friendly as well. I am a veteran and went to the school websites and Columbia Universiy have a General Studies program made for veterans and kind of techinqually free after aid 

(https://www.gs.columbia.edu/content/us-military-veterans)

It is very intresting to see. I added the ability to fully pay off student loans once completing your degree. I also searched for universities that are equipitable and affordable. I also added for first time/ first generation people going to college. Most on the schools are based near major cities which i am assuming becasue of the state tax benefits. 

I also found out that most ivy schools used to be military instiutions which makes sense for the veteran support being stronger than other school (https://www.militarytimes.com/news/education/2018/09/12/think-vets-like-you-dont-belong-at-ivy-league-schools-the-schools-may-disagree/)

Part 3: Program rankings

Using your two ranking systems, apply that methodology (as closely as you can) to individual programs of study. You can choose Bioinformatics programs if you want, but if you want to pick something else that's also fine. Display and contrast the results.

In [32]:
#https://www.forbes.com/video/6311598410112/the-top-5-colleges-in-the-us/?sh=46787428f2ab
#https://www.forbes.com/pictures/63062f99a431a1d1ebc1d377/top-25-lowest-debt/?sh=4cb895ae261a
#https://www.forbes.com/pictures/63067b2c97683f2a951b5fd4/1-harvard-university/?sh=1e2616cd43f3
#chrome-extension://efaidnbmnnnibpcajpcglclefindmkaj/https://collegescorecard.ed.gov/assets/InstitutionDataDocumentation.pdf

#College Score Card possible factors:
#state: "STABBR","median earing:'MD_EARN_WNE_P10',"GRAD_DEBT_MDN_SUPP",grad rates:'C150_4', retention rate:'RET_FT4'
#acceptance rate: ADM_RATE, median earing: MD_EARN_WNE_P10, retention rate: RET_FT4, grad rates: C150_4, loans: GRAD_DEBT_MDN_SUPP, fullpaid loans: BBRR2_FED_UG_PAIDINFULL_SUPP
#graduation: C150_4_POOLED_SUPP, student return: RET_FT4_POOLED_SUP, AVGFACSAL, TUITFTE, ADM_RATE_ALL
#grad rates:'C150_4', retention rate:'RET_FT4', median earing:'MD_EARN_WNE_P10', student_fac: 'STUFACR', Admin rate: 'ADM_RATE'



#Readind the data in and converting all PrivacySuppressed into Nan, I am using .dropna() for the after my function
mySchool = pd.read_csv("Most-Recent-Cohorts-Institution.csv", low_memory=False)
#replacing the PS to NaN
mySchool_ = mySchool.replace('PrivacySuppressed', np.nan)
#Dropping NaN values
mySchool_ = mySchool_.dropna(subset=['ADM_RATE','RET_FT4','BBRR2_FED_UG_PAIDINFULL_SUPP','MD_EARN_WNE_P10'
                                   ,'GRAD_DEBT_MDN_SUPP', 'VETERAN', 'PAR_ED_PCT_1STGEN', 'BBRR2_FED_UG_PAIDINFULL_SUPP',
                                  'TUITFTE'])



#function for the factors that are choosen
def scoreCard_2(row):
    #conveting admin rates into a floating (decimal accepting numbers) for var admin
    admin = float(row['ADM_RATE_ALL'])
    #converting retention into a float for var rten
    rten = float(row['RET_FT4']) 
    #converting total graduations into a float for var gard
    grad = float(row['C150_4_POOLED_SUPP'])
    #converting earnigns into a float var earnings
    earnings = float(row['MD_EARN_WNE_P10'])
    #converting debt into a float var debt
    debt = float(row['GRAD_DEBT_MDN_SUPP'])
    #converting current graduations to a float var grad2
    grant = float(row['BBRR2_FED_UG_PAIDINFULL_SUPP'])
    #converting student veteran population help into a float var stu
    stu = float(row['VETERAN'])
    #converting  first-generation on the FAFSA into a float var stu2
    stu2 = float(row['PAR_ED_PCT_1STGEN'])
    #converting student fully repaid debt into a float var stu
    debt2 = float(row['BBRR2_FED_UG_PAIDINFULL_SUPP'])
    #converting  first-generation on the FAFSA into a float var stu2
    stu3 = float(row['TUITFTE'])
    
    
    #creating the percentages for the values out of 1 (100%)
    uniPercent = {
        'admin': .1,
        'rten' : .1,
        'earnings' : .1,
        'debt' : .1,
        'stu2' : .1,
        'grad' : .1,
        'grant' : .1,
        'stu' : .1,
        'debt2' : .1,
        'stu3' : .1
    }
    
    #scoring system
    uniScore2 = (
        uniPercent['admin'] * admin
        + uniPercent['rten'] * rten
        + uniPercent['stu2'] * stu2
        + uniPercent['stu'] * stu
        + uniPercent['stu3'] * stu3
        + uniPercent['grad'] * grad) 
        #The ROI based on your earning potential and the amount of money/grants you can get
        #with amount of debt you can get subtracted by the ability to pay it off within 8 years
    + (uniPercent['earnings'] * earnings + uniPercent['grant'] * grant
        - (uniPercent['debt'] * debt - uniPercent['debt2'] * debt2))

    return uniScore2

#making sure the data within the file uses the scoreCard function to tally the points
mySchool_['uniScore2'] = mySchool_.apply(scoreCard_2, axis = 1)

#creating a var my_score to sort the values from the the file
my_score_ = mySchool_.sort_values(by = 'uniScore2', ascending=False)

#sorting the data and giving a ranking
my_score_['Rank'] = my_score_['uniScore2'].rank(ascending=False)

#var tamu scored the sorted by the scoring system with the 4 columns
nCU_ = my_score_[['Rank','uniScore2','INSTNM','STABBR', 'NPT4_PUB']]


#I had to unindex the column to be able to look for the school
nCU_[nCU_['STABBR']== 'NC']

Rank    uniScore2                                      INSTNM STABBR  \
2327   56.0  1710.544109                         Campbell University     NC   
4613  118.0  1375.042484        Johnson & Wales University-Charlotte     NC   
2390  190.0  1116.242214  North Carolina State University at Raleigh     NC   
2386  283.0   771.357599   University of North Carolina at Charlotte     NC   
2392  287.0   760.751265     University of North Carolina Wilmington     NC   
2343  297.0   740.963784                    East Carolina University     NC   
2315  324.0   678.264085                Appalachian State University     NC   
2383  325.0   673.716095       North Carolina A & T State University     NC   
2387  336.0   641.358470  University of North Carolina at Greensboro     NC   
2434  408.0   390.426109                 Western Carolina University     NC   
2395  419.0   325.940742    University of North Carolina at Pembroke     NC   
2348  428.0   277.722077               Fayetteville State University     NC   
5805    NaN          NaN       Chamberlain University-North Carolina     NC   

      NPT4_PUB  
2327       NaN  
4613       NaN  
2390   16411.0  
2386   17634.0  
2392   19136.0  
2343   16550.0  
2315   13989.0  
2383   10563.0  
2387   11629.0  
2434   13924.0  
2395   10792.0  
2348    6969.0  
5805       NaN

In [31]:
#https://www.forbes.com/video/6311598410112/the-top-5-colleges-in-the-us/?sh=46787428f2ab
#https://www.forbes.com/pictures/63062f99a431a1d1ebc1d377/top-25-lowest-debt/?sh=4cb895ae261a
#https://www.forbes.com/pictures/63067b2c97683f2a951b5fd4/1-harvard-university/?sh=1e2616cd43f3
#chrome-extension://efaidnbmnnnibpcajpcglclefindmkaj/https://collegescorecard.ed.gov/assets/InstitutionDataDocumentation.pdf

#College Score Card possible factors:
#state: "STABBR","median earing:'MD_EARN_WNE_P10',"GRAD_DEBT_MDN_SUPP",grad rates:'C150_4', retention rate:'RET_FT4'
#acceptance rate: ADM_RATE, median earing: MD_EARN_WNE_P10, retention rate: RET_FT4, grad rates: C150_4, loans: GRAD_DEBT_MDN_SUPP, fullpaid loans: BBRR2_FED_UG_PAIDINFULL_SUPP
#graduation: C150_4_POOLED_SUPP, student return: RET_FT4_POOLED_SUP, AVGFACSAL, TUITFTE, ADM_RATE_ALL
#grad rates:'C150_4', retention rate:'RET_FT4', median earing:'MD_EARN_WNE_P10', student_fac: 'STUFACR', Admin rate: 'ADM_RATE'



#Readind the data in and converting all PrivacySuppressed into Nan, I am using .dropna() for the after my function
mySchool = pd.read_csv("Most-Recent-Cohorts-Institution.csv", low_memory=False)
#replacing the PS to NaN
mySchool_ = mySchool.replace('PrivacySuppressed', np.nan)
#Dropping NaN values
mySchool_ = mySchool_.dropna(subset=['ADM_RATE','RET_FT4','BBRR2_FED_UG_PAIDINFULL_SUPP','MD_EARN_WNE_P10'
                                   ,'GRAD_DEBT_MDN_SUPP', 'VETERAN', 'PAR_ED_PCT_1STGEN', 'BBRR2_FED_UG_PAIDINFULL_SUPP',
                                  'TUITFTE'])



#function for the factors that are choosen
def scoreCard_2(row):
    #conveting admin rates into a floating (decimal accepting numbers) for var admin
    admin = float(row['ADM_RATE_ALL'])
    #converting retention into a float for var rten
    rten = float(row['RET_FT4']) 
    #converting total graduations into a float for var gard
    grad = float(row['C150_4_POOLED_SUPP'])
    #converting earnigns into a float var earnings
    earnings = float(row['MD_EARN_WNE_P10'])
    #converting debt into a float var debt
    debt = float(row['GRAD_DEBT_MDN_SUPP'])
    #converting current graduations to a float var grad2
    grant = float(row['BBRR2_FED_UG_PAIDINFULL_SUPP'])
    #converting student veteran population help into a float var stu
    stu = float(row['VETERAN'])
    #converting  first-generation on the FAFSA into a float var stu2
    stu2 = float(row['PAR_ED_PCT_1STGEN'])
    #converting student fully repaid debt into a float var stu
    debt2 = float(row['BBRR2_FED_UG_PAIDINFULL_SUPP'])
    #converting  first-generation on the FAFSA into a float var stu2
    stu3 = float(row['TUITFTE'])
    
    
    #creating the percentages for the values out of 1 (100%)
    uniPercent = {
        'admin': .1,
        'rten' : .1,
        'earnings' : .1,
        'debt' : .1,
        'stu2' : .1,
        'grad' : .1,
        'grant' : .1,
        'stu' : .1,
        'debt2' : .1,
        'stu3' : .1
    }
    
    #scoring system
    uniScore2 = (
        uniPercent['admin'] * admin
        + uniPercent['rten'] * rten
        + uniPercent['stu2'] * stu2
        + uniPercent['stu'] * stu
        + uniPercent['stu3'] * stu3
        + uniPercent['grad'] * grad) 
        #The ROI based on your earning potential and the amount of money/grants you can get
        #with amount of debt you can get subtracted by the ability to pay it off within 8 years
    + (uniPercent['earnings'] * earnings + uniPercent['grant'] * grant
        - (uniPercent['debt'] * debt - uniPercent['debt2'] * debt2))

    return uniScore2

#making sure the data within the file uses the scoreCard function to tally the points
mySchool_['uniScore2'] = mySchool_.apply(scoreCard_2, axis = 1)

#creating a var my_score to sort the values from the the file
my_score_ = mySchool_.sort_values(by = 'uniScore2', ascending=False)

#sorting the data and giving a ranking
my_score_['Rank'] = my_score_['uniScore2'].rank(ascending=False)

#var tamu scored the sorted by the scoring system with the 4 columns
tamu_ = my_score_[['Rank','uniScore2','INSTNM','STABBR', 'NPT4_PUB']]


#I had to unindex the column to be able to look for the school
tamu_[tamu_['INSTNM']== 'Texas A & M University-College Station']


#tamu_score[tamu_['CIPCODE']] == 'Biology' I could not find the subject for the field of study


Rank    uniScore2                                  INSTNM STABBR  \
3286  164.0  1197.869967  Texas A & M University-College Station     TX   

      NPT4_PUB  
3286   18919.0

I tried to look up Field of Study for Bioinformatics with 'CIPCODE' I even tried looking up Biology and could not get far so I choose to campare the how affordable schools are in NC within my rankings compared to TAMU. I and I noticed how the ranking changed from the rank in Part 4 from 38 to 164. Which puts Johnson and Wales/Campbell ahead of TAMU ranking wise. Campbell college cost does not appear to show neither. I did notice the cost for tutition for the most part in NC schools are close if not better than TAMU besides UNCW. I was suprised to not see the IVYs of the south in DUKE and UNC Chapel Hill made it to the list.

Part 4: Summarizing results

Finally, pick an institution that matters to you in some way. UNC Charlotte or where you did your undergrad are good choices, but you don't necessarily have to choose those. Where did that school fall in your ranking systems? If it was low in the rankings, what factors caused it to be? What steps could that institution take to improve its rankings? You can use past data as well to show how that institution is improving or failing to improve.

In [226]:
#https://www.forbes.com/video/6311598410112/the-top-5-colleges-in-the-us/?sh=46787428f2ab
#https://www.forbes.com/pictures/63062f99a431a1d1ebc1d377/top-25-lowest-debt/?sh=4cb895ae261a
#https://www.forbes.com/pictures/63067b2c97683f2a951b5fd4/1-harvard-university/?sh=1e2616cd43f3
#chrome-extension://efaidnbmnnnibpcajpcglclefindmkaj/https://collegescorecard.ed.gov/assets/InstitutionDataDocumentation.pdf

#College Score Card possible factors:
#state: "STABBR","median earing:'MD_EARN_WNE_P10',"GRAD_DEBT_MDN_SUPP",grad rates:'C150_4', retention rate:'RET_FT4'
#acceptance rate: ADM_RATE, median earing: MD_EARN_WNE_P10, retention rate: RET_FT4, grad rates: C150_4, loans: GRAD_DEBT_MDN_SUPP, fullpaid loans: BBRR2_FED_UG_PAIDINFULL_SUPP
#graduation: C150_4_POOLED_SUPP, student return: RET_FT4_POOLED_SUP, AVGFACSAL, TUITFTE, ADM_RATE_ALL
#grad rates:'C150_4', retention rate:'RET_FT4', median earing:'MD_EARN_WNE_P10', student_fac: 'STUFACR', Admin rate: 'ADM_RATE'



#Readind the data in and converting all PrivacySuppressed into Nan, I am using .dropna() for the after my function
mySchool = pd.read_csv("Most-Recent-Cohorts-Institution.csv", low_memory=False)
#replacing the PS to NaN
mySchool = mySchool.replace('PrivacySuppressed', np.nan)
#Dropping NaN values
mySchool = mySchool.dropna(subset=['ADM_RATE','RET_FT4','BBRR2_FED_UG_PAIDINFULL_SUPP','MD_EARN_WNE_P10'
                                   ,'GRAD_DEBT_MDN_SUPP', 'VETERAN', 'PAR_ED_PCT_1STGEN', 'BBRR2_FED_UG_PAIDINFULL_SUPP',
                                  'TUITFTE'])



#function for the factors that are choosen
def scoreCard2(row):
    #conveting admin rates into a floating (decimal accepting numbers) for var admin
    admin = float(row['ADM_RATE_ALL'])
    #converting retention into a float for var rten
    rten = float(row['RET_FT4']) 
    #converting total graduations into a float for var gard
    grad = float(row['C150_4_POOLED_SUPP'])
    #converting earnigns into a float var earnings
    earnings = float(row['MD_EARN_WNE_P10'])
    #converting debt into a float var debt
    debt = float(row['GRAD_DEBT_MDN_SUPP'])
    #converting current graduations to a float var grad2
    grant = float(row['BBRR2_FED_UG_PAIDINFULL_SUPP'])
    #converting student veteran population help into a float var stu
    stu = float(row['VETERAN'])
    #converting  first-generation on the FAFSA into a float var stu2
    stu2 = float(row['PAR_ED_PCT_1STGEN'])
    #converting student fully repaid debt into a float var stu
    debt2 = float(row['BBRR2_FED_UG_PAIDINFULL_SUPP'])
    #converting  first-generation on the FAFSA into a float var stu2
    stu3 = float(row['TUITFTE'])
    
    
    #creating the percentages for the values out of 1 (100%)
    uniPercent = {
        'admin': .1,
        'rten' : .1,
        'earnings' : .1,
        'debt' : .1,
        'stu2' : .1,
        'grad' : .1,
        'grant' : .1,
        'stu' : .1,
        'debt2' : .1,
        'stu3' : .1
    }
    
    #scoring system
    uniScore2 = (
        uniPercent['admin'] * admin
        + uniPercent['rten'] * rten
        + uniPercent['stu2'] * stu2
        + uniPercent['stu'] * stu
        + uniPercent['stu3'] * stu3
        + uniPercent['grad'] * grad) 
        + 
        #The ROI based on your earning potential and the amount of money/grants you can get
        #with amount of debt you can get subtracted by the ability to pay it off within 8 years
        (uniPercent['earnings'] * earnings + uniPercent['grant'] * grant
        - (uniPercent['debt'] * debt - uniPercent['debt2'] * debt2))

    return uniScore2

#making sure the data within the file uses the scoreCard function to tally the points
mySchool['uniScore2'] = mySchool.apply(scoreCard2, axis = 1)

#creating a var my_score to sort the values from the the file
my_score = mySchool.sort_values(by = 'uniScore2', ascending=False)

#sorting the data and giving a ranking
my_score['Rank'] = my_score['uniScore2'].rank(ascending=False)

#var tamu scored the sorted by the scoring system with the 4 columns
tamu = my_score[['Rank','uniScore2','INSTNM','STABBR']]


#I had to unindex the column to be able to look for the school
tamu[tamu['INSTNM']== 'Texas A & M University-College Station']


Rank    uniScore2                                  INSTNM STABBR
3286  38.0  6028.006443  Texas A & M University-College Station     TX

TAMU is ranked 38 out of 1492, which makes it in the top 2% of colleges for my lists. Moving from the city to College Station was a huge difference but it was an intresting time none the less. The school does lack demographic diversity (https://accountability.tamu.edu/All-Metrics/Mixed-Metrics/Student-Demographics) but for the most part everyone was welcoming. The resources were great. There is a website (https://www.applytexas.org/) that allows you to apply to colleges within Texas application free. You are able to travel to Houston, Austin, Dallas, San Antino and other diverse cities. Colorado is a cheap flight away. More people from California were and are moving there which makes it more expensive but more city people friendly. Currently I have a friend that is teaching there and a old work colleuge that graduated from its MD program

The school was a military school and still holds some of the traditions til this day. They are welcoming to everyone regardless of background. The school has a great veteran and military network that makes things accessible to everyone. The only problem is the poitical climate of the state government but that is appearent in most states. 